tests/backup/backup_delete_test.go

```diff
@@ -394,8 +396,41 @@ var _ = Describe("{DeleteIncrementalBackupsAndRecreateNew}", func() {
 				for _, vol := range bkpInspectResponse.GetBackup().GetVolumes() {
 					backupId := vol.GetBackupId()
 					log.InfoD(fmt.Sprintf("Backup Name: %s; BackupID: %s", incrementalBackupName, backupId))
-					dash.VerifyFatal(strings.Contains(backupId, "incr"), true,
-						fmt.Sprintf("Check if the backup %s is incremental or not ", incrementalBackupName))
+					if strings.Contains(backupId, "incr") {
+						dash.VerifyFatal(strings.Contains(backupId, "incr"), true,
+							fmt.Sprintf("Check if the backup %s is incremental or not ", incrementalBackupName))
+					} else {
+						// Attempting to take backups and checking if they are incremental or not
+						// as the original incremental backup which we took has taken a full backup this is mostly
+						// because CloudSnap is taking full backup instead of incremental backup as it's hitting one of
+						// the if else condition in CloudSnap which forces it to take full instead of incremental backup
+						log.InfoD("New backup wasn't an incremental backup hence recreating new backup")
+						for maxBackupsBeforeIncremental := 0; maxBackupsBeforeIncremental < 3; maxBackupsBeforeIncremental++ {
+							log.InfoD(fmt.Sprintf("Recreate incremental backup iternation: %d", maxBackupsBeforeIncremental))
+							for _, namespace := range bkpNamespaces {
+								// Create a new incremental backups
+								incrementalBackupName = fmt.Sprintf("%s-%s-%v", "incremental-backup", namespace, time.Now().Unix())
+								incrementalBackupNames = append(incrementalBackupNames, incrementalBackupName)
+								err = CreateBackup(incrementalBackupName, SourceClusterName, customBackupLocationName, backupLocationUID, []string{namespace},
+									labelSelectors, orgID, clusterUid, "", "", "", "", ctx)
+								dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying incremental backup [%s] creation", incrementalBackupName))
+
+								// Check if they are incremental or not
+								bkpUid, err = backupDriver.GetBackupUID(ctx, incrementalBackupName, orgID)
```

Comment:
As discussed, please check why do we need backup UID

Reply:
Done